src/test/java/com/ea/restaurant/service/impl/UserServiceE2ETest.java

```diff
@@ -0,0 +1,88 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.repository.UserRepository;
+import com.ea.restaurant.service.UserService;
+import com.ea.restaurant.service.impl.utils.fixtures.UserFixture;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@SpringBootTest
+public class UserServiceE2ETest {
+
+  @Autowired private JdbcTemplate jdbcTemplate;
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private UserService userService;
+
+  private User userToTest1;
+  private User userToTest2;
+
+  @BeforeEach
+  void setUp() {
+    jdbcTemplate.execute("TRUNCATE TABLE users CASCADE");
+    jdbcTemplate.execute("ALTER SEQUENCE users_id_seq RESTART");
+    userService = new UserServiceImpl(userRepository);
+    userToTest1 = UserFixture.buildUser("ea", "pg", "eapg", "1234");
+    userToTest2 = UserFixture.buildUser("ep", "gp", "ea001", "abcd1234");
+  }
+
+  @Test
+  public void createUserSuccessfully() {
+    User userSaved = this.userService.create(this.userToTest1);
+    Assertions.assertEquals(this.userToTest1, userSaved);
+  }
+
+  @Test
+  public void findEnabledByIdSuccessfully() {
+    User userSaved = this.userService.create(this.userToTest1);
+    Optional<User> userReturned = this.userService.findEnabledById(userSaved.getId());
+    Assertions.assertEquals(userReturned, Optional.of(this.userToTest1));
+  }
+
+  @Test
+  public void findEnabledByUsernameSuccessfully() {
+    this.userService.create(this.userToTest2);
+    Optional<User> userReturned =
+        this.userService.findEnabledByUsername(this.userToTest2.getUsername());
+    Assertions.assertEquals(this.userToTest2.getUsername(), userReturned.get().getUsername());
+  }
+
+  @Test
+  public void findAllEnabledSuccessfully() {
+    this.userService.create(this.userToTest1);
+    this.userService.create(this.userToTest2);
+    List<User> users = Arrays.asList(this.userToTest1, this.userToTest2);
+    List<User> usersReturned = this.userService.findAllEnabled();
+    Assertions.assertEquals(usersReturned, users);
+  }
+
+  @Test
+  public void deleteUserByIdSuccessfully() {
+    User user1 = this.userService.create(this.userToTest1);
+    User user2 = this.userService.create(this.userToTest2);
+    user1.setEntityStatus(Status.DELETED);
```

Comment:
Do a `findById`  and delete the returned user

Reply:
You don't need to set this to deleted, the delete method should do it for you already